### PR TITLE
testing state

### PR DESCRIPTION
**Pull request by AKobets**
Adding some test for StateStore

### CODE REVIEW

File: test/state/store.test.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		sandbox = sinon.createSandbox();
+	});
+
+	afterEach(() => {
+		emptyRecycleBin();
+		sandbox.restore();
+	});
+
+	describe('stateFactory', () => {
+		/*
```
Comment (AKobets) [inline test/state/store.test.js:43]: will remove all Testing notes comments that not needed. 
For now I use them to keep track of my questions while learning code and creating tests

File: test/state/store.test.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+			it(`should retun undefined when created with id:  ${JSON.stringify(test.id)}`, async() => {
+				const state = await createTestState(undefined, 'token');
+				expect(state).to.be.equal(undefined);
+			});
+		});
+
+		describe('integration with token', () => {
+			it('can create state with TOKEN_COOKIE', async() => {
+				const state = await createTestState('id', TOKEN_COOKIE);
+				verifyState(state, 'id', TOKEN_COOKIE);
```
Comment (AKobets) [inline test/state/store.test.js:89]: state could be created with TOKEN_COOKIEbut could not be r etrieved later with SateStore class method get()

Comment (zinabat) [reply test/state/store.test.js:89]: Excellent - new tests are finding bugs!

Comment (AKobets) [reply test/state/store.test.js:89]: same problem when cerating state with undefined token, the state is created but get throws. See failing tests in git actions logs
```
 ✖ can create state with TOKEN_COOKIE
        ✔ can create state with function token
        ✔ can create state with function that returns promise
        ✖ can create new state with undefined token
```

**Comment by AKobets** [comment]
there is another PR in draft state that tests most recent implementation.

**Pull request closed without merging**